frontend/src/modules/notification/actions.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react"
+import { AppDispatch, RootState } from "../../store"
+import { NOTIFICATION_REMOVE, NotificationAction as NotificationAction, NotificationID, NotificationProps, NotificationRemoveAction as NotificationRemoveAction, NOTIFICATION, NotificationType } from "./models"
+import { notification } from "antd"
+
+type AntdNotificationAPI = keyof Pick<typeof notification, 'success' | 'info' | 'warning' | 'error'>
+
+// converts our NotificationType to the function names in the Antd Notification API
+const convertToAntdNotificationAPI: { [key in NotificationType]: AntdNotificationAPI } = {
+    [NotificationType.SUCCESS]: 'success',
+    [NotificationType.INFO]: 'info',
+    [NotificationType.WARNING]: 'warning',
+    [NotificationType.ERROR]: 'error'
+}
+
+const hasNotification = (state: RootState, id: NotificationID) => {
+    return state.notifications.some((notification) => notification.id === id)
+}
+
+export const notify = (id: NotificationID, props: NotificationProps) => (dispatch: AppDispatch, getState: () => RootState) => {
+    if (hasNotification(getState(), id)) {
+        dispatch(closeNotification(id))
+    }
+
+    notification[convertToAntdNotificationAPI[props.type]]({
+        message: props.title,
+        description: <pre style={{ fontSize: '0.8em', whiteSpace: 'pre-wrap' }}>{props.description}</pre>,
+        duration: props.duration ?? 4.5, // 4.5 is the default duration in antd
+        key: id,
+        onClose: () => dispatch(closeNotification(id))
+    });
+
+    dispatch<NotificationAction>({
+        type: NOTIFICATION,
+        id,
+        props: props
+    })
+}
+
+export const closeNotification = (id: NotificationID) => (dispatch: AppDispatch, getState: () => RootState) => {
+    if (!hasNotification(getState(), id)) {
+        return;
+    }
+    dispatch<NotificationRemoveAction>({
+        type: NOTIFICATION_REMOVE,
+        id
+    })
+
+    // notification has onClose config which dispatches closeNotification.
+    // Closing after dispatch should prevent duplicate remove action.
+    notification.close(id)
+}
+
+export const showNotification = (
```

Comment:
I'm not sure this action is a good idea. Mixing the title and ID might be a poor practice. It might be better to simply use `notify` to pop alerts. It might be nice to have `errorNotification`, `warningNotification`, `infoNotification`, `successNotification` actions as a short cut:
```
export const errorNotification = (id: NotificationID, title: NotificationProps['title'], description: NotificationProps['description']) {
    return notify(id, {
        title,
        description,
        type: NotificationType.ERROR
    })
}
```

Reply:
Hopefully you like my approach to writing those extra actions ;)